app/models/host.rb

```diff
@@ -12,4 +14,23 @@ class Host < ApplicationRecord
   has_many :profiles, through: :profile_hosts, source: :profile
 
   validates :name, presence: true, uniqueness: { scope: :account_id }
+
+  class << self
+    def filter_by_compliance(key, operator, value)
+      ids = Host.includes(:profiles).all.select do |host|
+        host.compliant.values.all?(ActiveModel::Type::Boolean.new.cast(value))
+      end.pluck(:id).map { |id| "'#{id}'" }
+      return { :conditions => "1=0" } if ids.empty?
+      { :conditions => "hosts.id IN(#{ids.join(',')})" }
+    end
+
+    def filter_by_compliance_score(key, operator, range)
```

Comment:
This is how I expected it to work:

```rb
> Host.search_for("compliance_score < 100")
> Host.search_for("compliance_score < 99 and compliance_score > 97")
```

This feels a bit awkward (am I using it right?):

```rb
> Host.search_for("compliance_score = 97-99")
```